### ISSUE TITLE
Ray colocator test failure with nightly tests

### ISSUE THREAD

**Issue by tgaddair** [horovod/horovod#2574]
```

______________________________ test_colocator_gpu ______________________________
--
  |  
  | tmpdir = local('/tmp/pytest-of-root/pytest-0/test_colocator_gpu0')
  | ray_start_4_cpus_4_gpus = {'metrics_export_port': 45147, 'node_id': 'd8e6f25f505063efe553607a1a5706ad33345eb9', 'node_ip_address': '172.18.0.2', 'object_store_address': '/tmp/ray/session_2021-01-05_19-14-35_284083_496/sockets/plasma_store', ...}
  |  
  | @pytest.mark.skipif(
  | torch.cuda.device_count() < 4, reason='GPU colocator test requires 4 GPUs')
  | @pytest.mark.skipif(
  | not torch.cuda.is_available(), reason='GPU colocator test requires CUDA')
  | def test_colocator_gpu(tmpdir, ray_start_4_cpus_4_gpus):
  | SetColocator = NodeColocator.options(num_cpus=4, num_gpus=4)
  | colocator = SetColocator.remote(
  | node_rank=0, num_slots=4, world_size=4, use_gpu=True)
  | colocator.create_workers.remote()
  | worker_handles = ray.get(colocator.get_workers.remote())
  | assert len(set(ray.get(
  | [h.hostname.remote() for h in worker_handles]))) == 1
  | resources = ray.available_resources()
  | ip_address = services.get_node_ip_address()
  | assert resources.get("CPU", 0) == 0, resources
  | assert resources.get("GPU", 0) == 0, resources
  |  
  | # TODO: https://github.com/horovod/horovod/issues/2438
  | # assert resources.get(f"node:{ip_address}", 0) == 1 - 4 * 0.01
  |  
  | all_envs = ray.get([h.env_vars.remote() for h in worker_handles])
  | >       assert len({ev["CUDA_VISIBLE_DEVICES"] for ev in all_envs}) == 1
  | E       assert 2 == 1
  | E         +2
  | E         -1
  |  
  | test_ray.py:126: AssertionError


```

https://buildkite.com/horovod/horovod/builds/4633#b1bc3903-7817-461a-a806-077f454d0d57/312-397

cc @richardliaw

**Comment by richardliaw** [comment]
Hmm, looking..

**Comment by tgaddair** [comment]
Fixed by #2575.
